### PR TITLE
perf: Misc perf improvements

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 8a116f7</samp>

This pull request fixes various code analysis warnings, optimizes some string and memory operations, and updates some syntax to use new C# features. The changes affect files related to data binding, XAML parsing, media player, navigation view, maps, and Skia graphics. The changes improve the performance, reliability, and readability of the code.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Wpf/Extensions/TypeConversionExtensions.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ private static System.Windows.Media.GradientStop ToWpfGradientStop(this Windows.
 	{
 		if (imageSource is Windows.UI.Xaml.Media.Imaging.BitmapImage bitmapSource)
 		{
-			new System.Windows.Media.Imaging.BitmapImage(bitmapSource.UriSource);
+			return new System.Windows.Media.Imaging.BitmapImage(bitmapSource.UriSource);
```
Comment (Youssef1313) [inline src/Uno.UI.Runtime.Skia.Wpf/Extensions/TypeConversionExtensions.cs:58]: 😱

File: src/Uno.UI/UI/Xaml/Maps/MapInputEventArgs.cs [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+public partial class MapInputEventArgs : DependencyObject
 {
-	public partial class MapInputEventArgs : DependencyObject
-	{
-		public Windows.Devices.Geolocation.Geopoint Location { get; }
-		public Windows.Foundation.Point Position { get; }
+	public Geopoint Location { get; }
+	public Point Position { get; }
 
-		public MapInputEventArgs() : base() => new MapInputEventArgs(null, new Windows.Foundation.Point());
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Maps/MapInputEventArgs.cs:9]: This was creating a MapInputEventArgs that is thrown away immediately :(

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Media/Animation/RepeatBehavior.cs [hunk 3]
```diff
@@ -109,9 +109,9 @@ public string ToString(IFormatProvider provider)
 			{
 				return Forever;
 			}
-			else if (str.EndsWith("x", StringComparison.InvariantCultureIgnoreCase))
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Media/Animation/RepeatBehavior.cs:112]:
```csharp
    [Params("Abc", "Abcx", "AbcX")]
    public string MyValue { get; set; }

    [Benchmark(Baseline = true)]
    public bool StringComparisonInvariantIgnoreCase()
    {
        return MyValue.EndsWith("x", StringComparison.InvariantCultureIgnoreCase);
    }

    [Benchmark]
    public bool StringComparisonOrdinalIgnoreCase()
    {
        return MyValue.EndsWith("x", StringComparison.OrdinalIgnoreCase);
    }

    [Benchmark]
    public bool CharComparison()
    {
        return MyValue.EndsWith('x') || MyValue.EndsWith('X');
    }
```

image

Quote interesting.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Media/Animation/RepeatBehavior.cs (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Media/Animation/RepeatBehavior.cs:112]: Very interesting!

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.0

**Pull request merged**